tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -558,13 +558,34 @@ static void ExtendDestinationModel(std::span<const RFieldDescriptor *> newFields
         (newFields.size() > 1 ? "them" : "it"), mergeData.fNumDstEntries);
 
    changeset.fAddedFields.reserve(newFields.size());
+   // First add all non-projected fields...
    for (const auto *fieldDesc : newFields) {
-      auto field = fieldDesc->CreateField(*mergeData.fSrcDescriptor);
-      if (fieldDesc->IsProjectedField())
+      if (!fieldDesc->IsProjectedField()) {
+         auto field = fieldDesc->CreateField(*mergeData.fSrcDescriptor);
+         changeset.AddField(std::move(field));
+      }
+   }
+   // ...then add all projected fields.
+   Internal::RProjectedFields::FieldMap_t fieldMap;
+   for (const auto *fieldDesc : newFields) {
+      if (fieldDesc->IsProjectedField()) {
```

Comment:
Tiny style point: In such a case (long `if` block as the only element of a `for` block), I'd find
```
    if (!fieldDesc->IsProjectedField())
      continue;

   ...
```
slightly more readable.  Feel free to ignore though.